# Task 2: Titanic Predictive Model Report

**Objective:** The goal of this task was to build a machine learning model to predict the survival of passengers on the Titanic based on features like age, gender, and passenger class.

**Model Performance:** The model achieved an accuracy of **81.01%**.

In [2]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

# Load your file
df = pd.read_csv('Titanic-Dataset.csv')

# Look at the data to make sure it loaded
print(df.head())

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  


In [3]:
# 1. Fill missing Age values with the average age
df['Age'] = df['Age'].fillna(df['Age'].mean())

# 2. Change 'Sex' (male/female) into numbers (0 and 1)
df['Sex'] = df['Sex'].map({'male': 0, 'female': 1})

# 3. Drop columns that don't help prediction (Names, Tickets, etc.)
df = df.drop(['Name', 'Ticket', 'Cabin', 'PassengerId', 'Embarked'], axis=1)

print("Data is cleaned!")

Data is cleaned!


**Approach and Steps Followed:**

**Data Cleaning:** Handled missing values in the 'Age' column by filling them with the mean.

**Feature Engineering:** Converted categorical data ('Sex') into numerical format (0 for male, 1 for female) so the model could process it.

**Feature Selection:** Dropped non-essential columns such as 'Name', 'Ticket', and 'PassengerId' to reduce noise.

**Model Selection:** Used Logistic Regression because it is an efficient algorithm for binary classification (Survived vs. Not Survived).

**Data Splitting:** Divided the dataset into a training set (80%) and a testing set (20%) to validate performance.

In [4]:
# X = Features (Age, Class, etc.), y = Target (Survived or not)
X = df.drop('Survived', axis=1)
y = df['Survived']

# Split: 80% for training, 20% for testing your model's brain
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Create the Model
model = LogisticRegression()

# Train the Model
model.fit(X_train, y_train)

print("Training complete!")

Training complete!


In [5]:
# Predict using the test data
predictions = model.predict(X_test)

# Calculate Accuracy
score = accuracy_score(y_test, predictions)
print(f"My Model Accuracy: {score * 100:.2f}%")

My Model Accuracy: 81.01%


**Model Performance:**

**Accuracy: 81.01%**

**Observation:** The high accuracy suggests that the chosen features (especially Gender and Class) are strong predictors of survival.

**Conclusion:**
The model successfully predicts passenger survival with over 81% accuracy. For future tasks, accuracy could be further improved by exploring non-linear models like Random Forests or by handling the 'Embarked' column more deeply.